module/actor/sheets/mixins/healMixin.js

```diff
@@ -0,0 +1,115 @@
+const DialogV2 = foundry.applications.api.DialogV2;
+
+export let healMixing = {
+    async _onHeal() {
+        const { actor } = this;
+        const { rec } = actor.system.coreStats;
+        const { hp, sta } = actor.system.derivedStats;
+        const actualWoundList = actor.system.critWounds.length;
+
+        let totalRec = Math.floor(rec.max / 2);
+
+        const dialogData = {
+            totalRec,
+            actor,
+            isResting: false,
+            isSterilized: false,
+            isHealingHand: false,
+            isHealingTent: false,
+            daysHealed: 1,
+            actualWoundList: actualWoundList
+        };
+
+        await new DialogV2({
+            window: { title: game.i18n.localize('WITCHER.Heal.dialogTitle') },
+            content: await renderTemplate('systems/TheWitcherTRPG/templates/dialog/heal/heal-rest.hbs', dialogData),
+            modal: true,
+            buttons: [
+                {
+                    action: 'Heal',
+                    label: game.i18n.localize('WITCHER.Heal.button'),
+                    callback: async () => {

```

Comment:
I would prefer to not have callback functions in the dialog. The dialog should provide the necessary data but not do the computation